src/elements/content-sidebar/Sidebar.js

```diff
@@ -60,46 +67,81 @@ const Sidebar = ({
     fileId,
     getPreview,
     getViewer,
+    hasActivityFeed,
+    hasDetails,
+    hasMetadata,
+    hasSkills,
+    isOpen,
     metadataSidebarProps,
     onVersionHistoryClick,
-    selectedView,
 }: Props) => (
-    <React.Fragment>
-        {selectedView === SIDEBAR_VIEW_DETAILS && (
-            <LoadableDetailsSidebar
-                key={fileId}
-                fileId={fileId}
-                onVersionHistoryClick={onVersionHistoryClick}
-                {...detailsSidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_DETAILS}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_SKILLS && (
-            <LoadableSkillsSidebar
-                key={file.id}
-                file={file}
-                getPreview={getPreview}
-                getViewer={getViewer}
-                startMarkName={MARK_NAME_JS_LOADING_SKILLS}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_ACTIVITY && (
-            <LoadableActivitySidebar
-                currentUser={currentUser}
-                file={file}
-                onVersionHistoryClick={onVersionHistoryClick}
-                {...activitySidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_ACTIVITY}
-            />
-        )}
-        {selectedView === SIDEBAR_VIEW_METADATA && (
-            <LoadableMetadataSidebar
-                fileId={fileId}
-                {...metadataSidebarProps}
-                startMarkName={MARK_NAME_JS_LOADING_METADATA}
-            />
-        )}
-    </React.Fragment>
+    <Switch>
+        <Route exact path="/">
+            <Redirect to={`/${SIDEBAR_VIEW_SKILLS}`} />
+        </Route>
+        <SidebarRoute
```

Comment:
This looks like it captures the precedence of the sidebar tabs, but does it handle the use case where the sidebar should stick to prior chosen tab if it exists when navigating files?

Reply:
Yes, since the "route" isn't changed during file collection navigation.

Reply:
Nice!